In [1]:
import os
import json
import numpy as np
import pandas as pd

In [2]:
FT_result_dir = os.path.join("..","results","EFO_0003884",)

## Load the evaluaiton results file, its mentioend as json
with open(os.path.join(FT_result_dir, "evaluation_results","pinnacle_seed=1","pinnacle_EFO_0003884_results.csv"), "r") as f:
    eval_results = json.load(f)
eval_results.keys()

dict_keys(['ap', 'auroc'])

In [3]:
eval_results["auroc"]
celltype_auroc_df = pd.DataFrame.from_dict(eval_results["auroc"], orient="index", columns=["auroc"])
celltype_auroc_df = celltype_auroc_df.reset_index().rename(columns={"index":"cell_type"}).sort_values("auroc", ascending=False)
celltype_auroc_df.to_csv(os.path.join(FT_result_dir,"celltype_rank_auroc.csv"),index=False)
celltype_auroc_df.head()

,cell_type,auroc
10,sperm,0.645614
3,hepatocyte,0.555417
1,fibroblast,0.536111
8,monocyte,0.500000
0,adipocyte,0.472015


In [4]:
eval_results["ap"]
celltype_ap_df = pd.DataFrame.from_dict(eval_results["ap"], orient="index", columns=["ap"])
celltype_ap_df = celltype_ap_df.reset_index().rename(columns={"index":"cell_type"}).sort_values("ap", ascending=False)
celltype_ap_df.to_csv(os.path.join(FT_result_dir,"celltype_rank_apr_at_k.csv"),index=False)
celltype_ap_df.head(10)

,cell_type,ap
8,monocyte,0.253859
1,fibroblast,0.220115
3,hepatocyte,0.173322
4,ionocyte,0.159925
10,sperm,0.143690
0,adipocyte,0.114139
6,macrophage,0.113708
7,melanocyte,0.091416
5,keratinocyte,0.087110
9,plasmablast,0.073327


In [ ]:
celltype_ap_df

In [7]:
eval_results["ap"]

{'acinar cell of salivary gland': 0.06590281026823697,
 'adipocyte': 0.08813898630993178,
 'adventitial cell': 0.08189455075791655,
 'alveolar fibroblast': 0.06991253662610666,
 'artery endothelial cell': 0.09036543443692327,
 'b cell': 0.2176469516911688,
 'basal cell': 0.07570984710719168,
 'basal cell of prostate epithelium': 0.02774304659576957,
 'bladder urothelial cell': 0.057156218363581555,
 'bronchial smooth muscle cell': 0.08466917234425791,
 'bronchial vessel endothelial cell': 0.10084383767460867,
 'capillary endothelial cell': 0.17494405657566808,
 'cardiac endothelial cell': 0.11218270105519869,
 'cardiac muscle cell': 0.09298546314409142,
 'cd141-positive myeloid dendritic cell': 0.18197576200396476,
 'cd1c-positive myeloid dendritic cell': 0.12496779069699251,
 'cd24 neutrophil': 0.12864612421379018,
 'cd4-positive helper t cell': 0.17514693445962087,
 'cd4-positive, alpha-beta memory t cell': 0.41872629682083107,
 'cd8-positive, alpha-beta cytokine secreting effector t

In [9]:
FT_result_dir = os.path.join("..","results","EFO_0000685","model_outputs","pinnacle_seed=1")
csv_filenames_in_all = [i for i in os.listdir(FT_result_dir) if "_all_" in i]
csv_filenames_in_all

['pinnacle_all_preds_acinar cell of salivary gland.csv',
 'pinnacle_all_preds_adipocyte.csv',
 'pinnacle_all_preds_adventitial cell.csv',
 'pinnacle_all_preds_alveolar fibroblast.csv',
 'pinnacle_all_preds_artery endothelial cell.csv',
 'pinnacle_all_preds_b cell.csv',
 'pinnacle_all_preds_basal cell.csv',
 'pinnacle_all_preds_basal cell of prostate epithelium.csv',
 'pinnacle_all_preds_bladder urothelial cell.csv',
 'pinnacle_all_preds_bronchial smooth muscle cell.csv',
 'pinnacle_all_preds_bronchial vessel endothelial cell.csv',
 'pinnacle_all_preds_capillary endothelial cell.csv',
 'pinnacle_all_preds_cardiac endothelial cell.csv',
 'pinnacle_all_preds_cardiac muscle cell.csv',
 'pinnacle_all_preds_cd141-positive myeloid dendritic cell.csv',
 'pinnacle_all_preds_cd1c-positive myeloid dendritic cell.csv',
 'pinnacle_all_preds_cd24 neutrophil.csv',
 'pinnacle_all_preds_cd4-positive helper t cell.csv',
 'pinnacle_all_preds_cd4-positive, alpha-beta memory t cell.csv',
 'pinnacle_all_pre

In [10]:
df = pd.read_csv(os.path.join(FT_result_dir, csv_filenames_in_all[0]))
df.head()

,y,preds,name,type
0,0.0,0.213222,EGLN3,train
1,0.0,0.128638,MAP3K1,test
2,0.0,0.116498,KCNQ1,test
3,0.0,0.106897,NUCB1,test
4,0.0,0.101982,RIC3,train


In [11]:
def gene_pred_extractor(csv_filename, gene_symbol="CDK6", BASE_PATH = FT_result_dir):
    df = pd.read_csv(os.path.join(BASE_PATH, csv_filename))
    cell_type_name = csv_filename.replace(".csv","").replace("pinnacle_all_preds_","")
    if gene_symbol in df["name"].values:
        score = df[df["name"] == gene_symbol]["preds"].values[0]
        return score, cell_type_name
    ## Cell does not have the gene
    else:
        return np.nan, cell_type_name

## use gene_pred_extractor with a custom gene_symbol and apply it to csv_filenames list. use map or apply to speed up things

gene_symbol = "JAK3"
results = list(map(lambda x: gene_pred_extractor(x, gene_symbol=gene_symbol), csv_filenames_in_all))
# convert the list which returns a tuple to a dataframe so that we can sort it
df_results = pd.DataFrame(results, columns=["score", "cell_type"])
## Sort by score in descending order
df_results = df_results.sort_values(by="score", ascending=False)
df_results.dropna(inplace=True)
df_results.reset_index(drop=True, inplace=True)
df_results.head()

,score,cell_type
0,0.236386,"cd8-positive, alpha-beta cytotoxic t cell"
1,0.209317,microglial cell
2,0.128264,naive regulatory t cell
3,0.126089,t follicular helper cell
4,0.112337,regulatory t cell


In [12]:
## Iterate through the filenames, load the CSV, add a column where the values will be the celltype name from the filename. Concatenate all the csvs
dfs = []
for csv_filename in csv_filenames_in_all:
    df = pd.read_csv(os.path.join(FT_result_dir, csv_filename))
    cell_type_name = csv_filename.replace(".csv","").replace("pinnacle_all_preds_","")
    df["cell_type"] = cell_type_name
    dfs.append(df)
df_all = pd.concat(dfs, ignore_index=True)

In [13]:
df_all.head()

,y,preds,name,type,cell_type
0,0.0,0.213222,EGLN3,train,acinar cell of salivary gland
1,0.0,0.128638,MAP3K1,test,acinar cell of salivary gland
2,0.0,0.116498,KCNQ1,test,acinar cell of salivary gland
3,0.0,0.106897,NUCB1,test,acinar cell of salivary gland
4,0.0,0.101982,RIC3,train,acinar cell of salivary gland


In [14]:
## Groupby the name, for each groupped dataframe, find the row with greatest preds. make a df with these rows alone
df_best = df_all.loc[df_all.groupby("name")["preds"].idxmax()]
df_best = df_best.sort_values(by="preds", ascending=False)
df_best.head(20)

,y,preds,name,type,cell_type
13442,1.0,0.505330,TUBB3,test,enterocyte of epithelium of large intestine
47574,1.0,0.428548,PRIM2,train,type i nk t cell
13443,0.0,0.417561,MT-CO2,test,enterocyte of epithelium of large intestine
5913,0.0,0.406750,HGS,test,"cd8-positive, alpha-beta cytotoxic t cell"
13444,0.0,0.406527,CDC42BPG,val,enterocyte of epithelium of large intestine
47575,0.0,0.395735,SLAMF7,test,type i nk t cell
47576,0.0,0.395100,SNRK,test,type i nk t cell
37079,0.0,0.392107,SLC25A13,train,paneth cell of epithelium of large intestine
26796,0.0,0.387649,RFK,test,macrophage
26797,0.0,0.376021,ATOX1,test,macrophage


	y	preds	name	type	cell_type
8400	1.0	0.544774	FGFR3	train	conjunctival epithelial cell
24030	0.0	0.529834	TNK1	test	large intestine goblet cell
24031	1.0	0.527707	MAPKAPK5	train	large intestine goblet cell
0	0.0	0.514637	SHMT1	test	acinar cell of salivary gland
1	0.0	0.504812	AZGP1	val	acinar cell of salivary gland
47574	1.0	0.497695	PRIM2	train	type i nk t cell
41710	0.0	0.479576	ALAS1	train	retinal ganglion cell
42613	0.0	0.465789	GJA1	train	schwann cell
16440	1.0	0.458395	POLA2	test	fibroblast of cardiac tissue
2919	1.0	0.455851	JAK2	train	bronchial smooth muscle cell